packages/loopring_v3/contracts/iface/IBlockVerifier.sol

```diff
@@ -44,11 +46,13 @@ contract IBlockVerifier
     /// @param onchainDataAvailability True if the block expects onchain
     ///        data availability data as public input, false otherwise
     /// @param blockSize The number of requests handled in the block
+    /// @param blockVersion The block version (i.e. which circuit version needs to be used)
     /// @return True if the block can be verified, false otherwise
     function canVerify(
-        uint8 blockType,
-        bool onchainDataAvailability,
-        uint16 blockSize
+        uint8  blockType,
+        bool   onchainDataAvailability,
+        uint16 blockSize,
+        uint8  blockVersion
```

Comment:
Can we use old number for data availability and even number for no-data availability?

Reply:
Or use 0 for no-data availability and non zero for data availability with different compression algorithms

Reply:
Not sure, this might make things more complicated than needed.

The versions can also be used to deprecate a block version. So for example a new block version can be added in IBlockVerifier and the old version can only be used for another ~1 week before no new blocks can be committed with that version (but they can still be verified after that deadline) . This allows us to update the circuits when needed without any interruption.